clusterloader2/pkg/measurement/manager.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package measurement
+
+import (
+	"sync"
+
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+// MeasurementManager manages all measurements calls and executions.
+type MeasurementManager struct {
+	clientSet    clientset.Interface
+	lock         sync.Mutex
+	// measurements is a map: methodName -> identifier -> Measurement.
+	// First level divides measurements into method classes, second into instances.
+	measurements map[string]map[string]Measurement
+}
+
+// CreateMeasurementManager creates new instance of MeasurementManager.
+func CreateMeasurementManager(clientSet clientset.Interface) *MeasurementManager {
+	return &MeasurementManager{
+		clientSet:    clientSet,
+		measurements: make(map[string]map[string]Measurement),
+	}
+}
+
+// Execute executes measurement based on provided identifier, methodName and params.
+func (mm *MeasurementManager) Execute(identifier string, methodName string, params map[string]interface{}) error {
+	m, err := mm.getMeasurement(identifier, methodName)
+	if err != nil {
+		return err
+	}
+	config := &MeasurementConfig{
+		ClientSet: mm.clientSet,
+		Params:    params,
+	}
+	return m.Execute(config)
+}
+
+func (mm *MeasurementManager) getMeasurement(identifier string, methodName string) (Measurement, error) {
```

Comment:
-> getMeasurementObject? (because getMeasurement sounds ambiguous - it could even mean getting a real measurement)

Reply:
Nit - could you reverse the argument order of the function? (to match key order in the map)

Reply:
`getMeasurementInstance` it is.
Order of parameters changed.